internal/pkg/version/version.go

```diff
@@ -11,6 +11,10 @@ const (
 	AppTemplateMinAlias = "v1.0.0"
 	// AppTemplateMinStaticSite is the minimum app version required to deploy a static site.
 	AppTemplateMinStaticSite = "v1.2.0"
+	// LegacyEnvTemplate is the version associated with the environment template before we started versioning.
+	LegacyEnvTemplate = "v0.0.0"
+	// EnvTemplateBootstrap is the version of an environment template that contains only bootstrap resources.
+	EnvTemplateBootstrap = "bootstrap"
```

Comment:
A small call out which I think is probably fine - 
When people run `env init`, we deploy an "environment template" that contains only the CFN execution role and the env manager role. These two roles are used when people run `env deploy`.

An "environment template" at this point contains only the two roles and doesn't not contain any actual environment resources. This template is on "bootstrap" at this point.

With the current implementation, we (probably) won't be able to detect a "downgrade" in the following case:
```
$ copilot-v1.30.0 env init # Deploy an template on "bootstrap".
$ copilot-v1.29.0 env deploy 
```

Which I think is probably fine in most of the cases - unless we've patched some important permissions two either CFNExecutionRole or EnvManagerRole on v1.30.0.

Reply:
Yeah I think it should be fine because `env init` won't function anyway.